src/DebugSql.php

```diff
@@ -143,6 +161,16 @@ public static function sqld(Query $query, $showValues = true, $showHtml = null,
         die(1);
     }
 
+    /**
+     * Checks whether the current environment is CLI based.
+     *
+     * @return bool
+     */
+    protected static function isCli()
+    {
+        return PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg';
+    }
```

Comment:
This could probably be a global function in cakephp.

Reply:
Maybe. The core seems to differentiate in some areas between `cli` only, and `cli || phpdbg`.